src/liboslexec/llvm_util.cpp

```diff
@@ -437,7 +437,369 @@ LLVM_Util::end_builder ()
+#if OSL_LLVM_VERSION >= 70
+    "sahf",
+#endif
+    "sse", "sse2", "sse3", "sse4.1", "sse4.2", "ssse3", "x87"
+    // , "xsave", "xsavec", "xsaveopt", "xsaves" // Save Processor Extended States, we don't use
+};
+
+
+
+static void
+get_required_cpu_features_for(TargetISA target, const char ** &features, int &feature_count)
+{
+    OSL_ASSERT(target > TargetISA::UNKNOWN && target < TargetISA::COUNT);
+
+    switch(target) {
+    case TargetISA::x64:
+        features = required_cpu_features_by_x64;
+        feature_count = std::extent<decltype(required_cpu_features_by_x64)>::value;
+        break;
+    case TargetISA::SSE4_2:
+        features = required_cpu_features_by_SSE4_2;
+        feature_count = std::extent<decltype(required_cpu_features_by_SSE4_2)>::value;
+        break;
+    case TargetISA::AVX:
+        features = required_cpu_features_by_AVX;
+        feature_count = std::extent<decltype(required_cpu_features_by_AVX)>::value;
+        break;
+    case TargetISA::AVX2:
+        features = required_cpu_features_by_AVX2;
+        feature_count = std::extent<decltype(required_cpu_features_by_AVX2)>::value;
+        break;
+    case TargetISA::AVX2_noFMA:
+        features = required_cpu_features_by_AVX2_noFMA;
+        feature_count = std::extent<decltype(required_cpu_features_by_AVX2_noFMA)>::value;
+        break;
+    case TargetISA::AVX512:
+        features = required_cpu_features_by_AVX512;
+        feature_count = std::extent<decltype(required_cpu_features_by_AVX512)>::value;
+        break;
+    case TargetISA::AVX512_noFMA:
+        features = required_cpu_features_by_AVX512_noFMA;
+        feature_count = std::extent<decltype(required_cpu_features_by_AVX512_noFMA)>::value;
+        break;
+    default:
+        OSL_ASSERT(0 && "incomplete required cpu features for target are not specified");
+    }
+    OSL_ASSERT(features);
+    OSL_ASSERT(feature_count > 0);
+}
+
+
+
+/*static*/ TargetISA
+LLVM_Util::lookup_isa_by_name(string_view target_name)
+{
+    OSL_DEV_ONLY(std::cout << "lookup_isa_by_name(" << target_name << ")" << std::endl);
+    TargetISA requestedISA = TargetISA::UNKNOWN;
+    if (target_name != NULL) {
+        for (int i = static_cast<int>(TargetISA::UNKNOWN); i < static_cast<int>(TargetISA::COUNT); ++i) {
+            if (OIIO::Strutil::iequals(target_name, target_isa_names[i])) {
+                requestedISA = static_cast<TargetISA>(i);
+                OSL_DEV_ONLY(std::cout << "REQUESTED ISA:" << target_isa_names[i] << std::endl);
+                break;
+            }
+        }
+        // NOTE: we are ignoring unrecognized target strings
+    }
+    return requestedISA;
+}
+
+
+
+const char*
+LLVM_Util::target_isa_name(TargetISA isa)
+{
+    return target_isa_names[static_cast<int>(isa)];
+}
+
+
+
+bool
+LLVM_Util::detect_cpu_features(string_view requested_isa_string,
+                               bool no_fma)
+{
+    TargetISA requestedISA = lookup_isa_by_name(requested_isa_string);
```

Comment:
I am wondering if maybe `requested_isa_string` parameter to detect_cpu_features, should just be `requestedISA `and make caller call` lookup_isa_by_name(requested_isa_string) `themselves?

Reply:
I have no preference. Would you like me to change it?

Reply:
Yes, I think so, it was bugging me in use, and now that you have busted out detect_cpu_features to its own function, it makes sense.

Reply:
OK, will do.

Reply:
Amended.

Also make llvmutil_test double check that what it's getting from these LLVM checks of the host ISA matches what OIIO thinks it's getting from cpuid queries.